bitarray/_bitarray.c

```diff
@@ -2945,8 +2912,8 @@ static PyTypeObject Bitarraytype = {
     0,                                        /* tp_compare */
     (reprfunc) bitarray_repr,                 /* tp_repr */
     0,                                        /* tp_as_number*/
-    0,                                        /* tp_as_sequence */
-    0,                                        /* tp_as_mapping */
+    &bitarray_as_sequence,                    /* tp_as_sequence */
+    &bitarray_as_mapping,                     /* tp_as_mapping */
```

Comment:
The comment about length() is true. On 32-bit systems Py_ssize_t is 32 bits, so for bitarrays with more than 2**32-1 elements the call to len() should fail. Good point though, I've added a check. (I can't test it though, since I don't have a 32 bit python around.) We can't do better in this case, but I think huge bitarrays on 32-bit systems are not a big deal.

Converting the remaining functions to C is a trade off -- there will be more code, but it will be faster. There's a smallish memory overhead from using a Python class that can't be eliminated until Python 3.5. Of the functions that are in Python now, 2 are deprecated and others seem not very useful. If we're OK with making incompatible changes, we can get ride of them, or move them out of the class.